lib/BasicAnalyses/CSVAliasAnalysis.cpp

```diff
@@ -0,0 +1,259 @@
+/// \file CSVAliasAnalysis.cpp
+/// \brief Decorate memory accesses with information about CSV aliasing, and
+///        optionally segregate direct stack accesses from all other memory
+///        accesses through additional alias information.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/BasicAnalyses/CSVAliasAnalysis.h"
+
+using namespace llvm;
+
+class SegregateDirectStackAccesses {
+  CSVAliasAnalysisInterface *CSVAA;
```

Comment:
Do you actually need this?
I'd like `SegregateDirectStackAccesses` to be completely separate from `CSVAliasAnalysis`.